modules/draco/src/lib/draco-parser.js

```diff
@@ -263,71 +260,54 @@ export default class DracoParser {
       throw new Error(message);
     }
 
-    const attributeType = DRACO_DATA_TYPE_TO_TYPED_ARRAY_MAP[dracoAttribute.data_type()];
+    const TypedArrayCtor = DRACO_DATA_TYPE_TO_TYPED_ARRAY_MAP[dracoAttribute.data_type()];
     const numComponents = dracoAttribute.num_components();
     const numPoints = dracoGeometry.num_points();
     const numValues = numPoints * numComponents;
 
-    let dracoArray;
-    let typedArray;
+    const byteLength = numValues * TypedArrayCtor.BYTES_PER_ELEMENT;
+    const dataType = this._getDracoDataType(TypedArrayCtor);
 
-    switch (attributeType) {
-      case Float32Array:
-        dracoArray = new this.draco.DracoFloat32Array();
-        decoder.GetAttributeFloatForAllPoints(dracoGeometry, dracoAttribute, dracoArray);
-        typedArray = new Float32Array(numValues);
-        break;
+    const ptr = this.draco._malloc(byteLength);
+    decoder.GetAttributeDataArrayForAllPoints(
+      dracoGeometry,
+      dracoAttribute,
+      dataType,
+      byteLength,
+      ptr
+    );
+    const typedArray = new TypedArrayCtor(this.draco.HEAPF32.buffer, ptr, numValues).slice();
```

Comment:
Slice seems to create unnecessary copy?

Reply:
Wrapped `this.draco.HEAPF32.buffer` have to be disposed. A copy is necessary